src/renderer/components/dialogs/SetupMessage.js

```diff
@@ -14,33 +13,60 @@ class SetupMessagePanel extends React.Component {
   constructor (props) {
     super(props)
     this.state = {
-      key: ''
+      key: Array(9).fill('')
     }
-    this.handleChange = this.handleChange.bind(this)
+    this.handleChangeKey = this.handleChangeKey.bind(this)
   }
 
-  handleChange (event) {
+  handleChangeKey (event) {
     // TODO: insert - automatically in between every 4 characters
     // TODO: lint the value for correct setup message format
-    this.setState({ key: event.target.value, message: false })
+    if (event.target.value < 0 || event.target.value > 9999) {
```

Comment:
What happens when the user inputs a string like `asdf`? 
If the value isn't a number, we should return false also. So here:
```js
var value = Number(event.target.value)
if (isNaN(value) || value < 0 || value > 9999)  return false
let updatedkey = this.state.key
updatedkey[Number(event.target.id)] = value
this.setState({ key: updatedkey })
```